modules/openapi-generator/src/main/resources/go/partial_header.mustache

```diff
@@ -13,5 +13,11 @@
  {{#infoEmail}}
  * Contact: {{{infoEmail}}}
  {{/infoEmail}}
+{{^withGoCodegenComment}}
  * Generated by: OpenAPI Generator (https://openapi-generator.tech)
```

Comment:
@grokify thanks for the PR. What about replacing this line directly with the following to avoid adding another option?
```
// Code generated by OpenAPI Generator (https://openapi-generator.tech); DO NOT EDIT.
```

Reply:
My original commit f31ecadf24a3e82596020793bb626d65f4736116 was a simple replace, however, in discussion with @etherealjoy, I agreed it's not ideal to remove the original behavior and that the original behavior should be the default. This is because this change trigger's Go's special linter handling for machine generated code and disables `golint` and thus `gometalinter`. The result is that:

1. Warnings for exported symbols without comments will be suppressed
2. Warnings for non-standard MixedCase symbol names will be suppressed

For exported symbol comments, it seems generally better to have comments than not since this are expected to be used by humans. Depending on the specification, this can be the majority of warnings. The cases where people indicated a preference for suppression of these warnings seem to be related to when there are no documentation resources available for a legacy / deprecated API. It doesn't seem like a legacy / deprecated API use case should be the default behavior.

MixedCase is a different issue since sometimes I personally feel MixedCase linter is too strict. I like things like `userId` and `serverUrl` because it makes the symbol name more easily machine parseable, but I recognize the standard in the Go community here.

There is a way to separate this behavior, but that involves having `golint` issue the warnings and then using `gometalinter` to selectively remove the warnings you are not interested in.

So, the thought is to make the comment a non-default option because it will:

1. allow users to see comment errors which should be addressed most of the time
1. allow more customized behavior using a combination of `golint` and `gometalinter`
1. allow a different behavior for scenarios like legacy / deprecated API use cases

Reply:
@grokify thanks for the explanation. My feedback was to avoid adding too many options for a generator. I'm ok with this new option given the reason you provided.